plugins/modules/pem_read.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2025, Pablo Rodriguez <pabrodri@redhat.com>
+# Apache License Version 2.0 (see LICENSE)
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: url_request
+short_description: Reads a PEM file (that can list many certs)
+extends_documentation_fragment:
+    - files
+
+description:
+- Reads a PEM file (that can list many certs)
+- OU or CN regexes can be provided to filter out the list
+
+author:
+  - Pablo Rodriguez (@pablintino)
+
+options:
+  path:
+    description:
+      - The path to the certificate file to be read
+    required: True
+    type: str
+  ou_filter:
+    description:
+      - Regex that, if given, used to filter the list of certs by OU.
+    required: False
+    type: str
+  cn_filter:
+    description:
+      - Regex that, if given, used to filter the list of certs by CN.
+    required: False
+    type: str
+
+"""
+
+EXAMPLES = r"""
+- name: Get pem certs from crt file
+  pem_read:
+    path: "/etc/ssl/certs/ca-certificates.crt"
+  register: _certs
+
+- name: Get pem certs from crt file by OU
+  pem_read:
+    path: "/etc/ssl/certs/ca-certificates.crt"
+    ou_filter: "Red Hat"
+  register: _certs2
+"""
+
+RETURN = r"""
+certs:
+    description: The list of PEM files
+    type: list
+    returned: returned request
+"""
+
+import re
+import typing
+
+from cryptography import x509
+from cryptography.x509.oid import NameOID
+from cryptography.hazmat.primitives import serialization
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def filter_certs(
+    input_certs: typing.List[x509.Certificate], ou_filter=None, cn_filter=None
+) -> typing.List[x509.Certificate]:
+    certs = []
+    for cert in input_certs:
+        ou = cert.subject.get_attributes_for_oid(NameOID.ORGANIZATIONAL_UNIT_NAME)
+        cn = cert.subject.get_attributes_for_oid(NameOID.COMMON_NAME)
+        if (
+            (ou and ou_filter and re.search(ou_filter, ou[0].value, re.IGNORECASE))
```

Comment:
With regards with the use of `re.IGNORECASE`, regex is case sensitive so when providing a regex string I would be surprised if the case was not honored as it's not mentioned in the README.

Ignore this if OU and CN are always upper or lower case 😆